### ISSUE TITLE
Socketcluster not working with path

### ISSUE THREAD

**Issue by sisso19** [SocketCluster/socketcluster#172]
I am trying out socketcluster and I have a problem that I am unable to figure out and is causing me quite the headache.

I am trying to create the socket server with a path to the server e.g /socket/cluster, but it does not seem to work.

When I have no path defined in neither the client nor the server, everything works fine. But when I add the path to both ends it fails and gives me this error on the client side.

    { [SocketProtocolError: Client connection establishment timed out]
      name: 'SocketProtocolError',
      message: 'Client connection establishment timed out',
      code: 4007 }

I am currently just testing this out so there is no major code behind this. This is the client

    var io = require("socketcluster-client");
    var socket = io.connect({
    	      "hostname": "localhost",
	      "port": 10001,
	      "path": "/socket/cluster",
	      "autoreconnect": true
	    });
    
    socket.on("connect", function(){
    	console.log("connected");
    });

    socket.on("error", function(err){
    	console.log("error", err);
    });

And this is the server.js

    var SocketCluster = require('socketcluster').SocketCluster;
    var socketCluster = new SocketCluster({
      workers: 1, // Number of worker processes
      brokers: 1, // Number of broker processes
      port: 10001, // The port number on which your server should listen
      appName: 'myApp', // A unique name for your app
    
      /* A JS file which you can use to configure each of your
       * workers/servers - This is where most of your backend code should go
       */
      workerController: __dirname + '/worker.js',
    
      /* JS file which you can use to configure each of your
       * brokers - Useful for scaling horizontally across multiple machines (optional)
       */
      brokerController: __dirname + '/broker.js',
    
      // Whether or not to reboot the worker in case it crashes (defaults to true)
      rebootWorkerOnCrash: true,
    
      // how long until the ping times out
      pingTimeout: 5000,
    
      // how often shold the ping happen
      pingInterval: 2000,
      path: "/socket/cluster"
    });

I am using version *4.3.4* of socketcluster and *4.3.14* of the socketcluster-client

**Comment by sisso19** [comment]
I have been looking more into this and this seems to be an issue with socketcluster server.
When I define a path in the socketcluster server.js the server I cannot see that the server takes any requests.

So by removing the path from the server, I was able to get requests on /socket/cluster but sockets are able to connect to the server via any path.

So this is still an issue, but I have worked my way around it for now.
